interface/forms/fee_sheet/new.php

```diff
@@ -1289,68 +1176,76 @@ function setJustify(seljust) {
-      echo "    <option value='" . attr($date0) . "'>" . xlt('Contraceptive services previously started') . "</option>\n";
-      echo "    <option value=''>" . xlt('None of the above') . "</option>\n";
-      echo "   </select>\n";
-      echo "&nbsp; &nbsp; &nbsp;\n";
+if ($fs->contraception_code && !$isBilled) {
+  // This will give the form save logic the associated contraceptive method.
+  echo "<input type='hidden' name='ippfconmeth' value='" . attr($fs->contraception_code) . "'>\n";
+
+  // If Contraception forms can be auto-created by the Fee Sheet we might need
+  // to ask if this is the client's first contraception at this clinic.
+  //
+  if ($GLOBALS['gbl_new_acceptor_policy'] == '1') {
+    $csrow = sqlQuery("SELECT COUNT(*) AS count FROM forms AS f WHERE " .
+      "f.pid = ? AND f.encounter = ? AND f.formdir = 'LBFccicon' AND " .
+      "f.deleted = 0", array($fs->pid, $fs->encounter));
+    // Do it only if a contraception form does not already exist for this visit.
+    // Otherwise assume that whoever created it knows what they were doing.
+    if ($csrow['count'] == 0) {
+      $date1 = substr($visit_row['date'], 0, 10);
+      $ask_new_user = false;
+      // Determine if this client ever started contraception with the MA.
+      // Even if only a method change, we assume they have.
+      // *************************************************************
+      //// But this version would be used if method changes don't count:
+      // $query = "SELECT f.form_id FROM forms AS f " .
+      //   "JOIN form_encounter AS fe ON fe.pid = f.pid AND fe.encounter = f.encounter " .
+      //   "JOIN lbf_data AS d1 ON d1.form_id = f.form_id AND d1.field_id = 'newmethod' " .
+      //   "LEFT JOIN lbf_data AS d2 ON d2.form_id = f.form_id AND d2.field_id = 'newmauser' " .
+      //   "WHERE f.formdir = 'LBFccicon' AND f.deleted = 0 AND f.pid = '$pid' AND " .
+      //   "(d1.field_value LIKE '12%' OR (d2.field_value IS NOT NULL AND d2.field_value = '1')) " .
+      //   "ORDER BY fe.date DESC LIMIT 1";
+      // *************************************************************
+      $query = "SELECT f.form_id FROM forms AS f " .
+        "JOIN form_encounter AS fe ON fe.pid = f.pid AND fe.encounter = f.encounter " .
+        "WHERE f.formdir = 'LBFccicon' AND f.deleted = 0 AND f.pid = ? " .
+        "ORDER BY fe.date DESC LIMIT 1";
+      $csrow = sqlQuery($query, array($fs->pid));
+      if (empty($csrow)) {
+        $ask_new_user = true;
+      }
+      if ($ask_new_user) {
+        echo "<select name='newmauser'>\n";
+        echo " <option value='2'>" . xlt('First Modern Contraceptive Use (Lifetime)') . "</option>\n";
+        echo " <option value='1'>" . xlt('First Modern Contraception at this Clinic (with Prior Contraceptive Use)') . "</option>\n";
+        echo " <option value='0'>" . xlt('Method Change at this Clinic') . "</option>\n";
```

Comment:
Nice catch! The function in FeeSheetHtml was created to replace this but the code here was never replaced.  I will deal with it.